src/main/java/org/zalando/nakadi/controller/CursorOperationsController.java

```diff
@@ -0,0 +1,202 @@
+                             final CursorConverter cursorConverter) {
+        this.cursorOperationsService = cursorOperationsService;
+        this.cursorConverter = cursorConverter;
+    }
+
+    @RequestMapping(path = "/event-types/{eventTypeName}/cursor-distances", method = RequestMethod.POST)
+    public ResponseEntity<?> getDistance(@PathVariable("eventTypeName") final String eventTypeName,
+                                         @Valid @RequestBody final List<CursorDistanceQuery> queries,
+                                         final Errors errors,
+                                         final NativeWebRequest request) {
+
+        if (errors.hasErrors()) {
+            return Responses.create(new ValidationProblem(errors), request);
+        }
+
+        final List<NakadiCursorDistanceQuery> domainQueries = queries.stream()
+                .map(this.toCursorDistanceQuery(eventTypeName))
+                .collect(Collectors.toList());
+
+        final List<NakadiCursorDistanceResult> domainResult = cursorOperationsService.calculateDistance(domainQueries);
+
+        final List<CursorDistanceResult> viewResult = domainResult.stream().map(this::toCursorDistanceResult)
+                .collect(Collectors.toList());
+
+        return status(OK).body(viewResult);
+    }
+
+    @RequestMapping(path = "/event-types/{eventTypeName}/shifted-cursors", method = RequestMethod.POST)
+    public ResponseEntity<?> moveCursors(@PathVariable("eventTypeName") final String eventTypeName,
+                                         @Valid @RequestBody final List<ShiftedCursor> cursors,
+                                         final Errors errors,
+                                         final NativeWebRequest request) {
+
+        if (errors.hasErrors()) {
+            return Responses.create(new ValidationProblem(errors), request);
+        }
+
+        final List<ShiftedNakadiCursor> domainCursor = cursors.stream()
+                .map(this.toShiftedNakadiCursor(eventTypeName))
+                .collect(Collectors.toList());
+
+        final List<NakadiCursor> domainResultCursors = cursorOperationsService.unshiftCursors(domainCursor);
+
+        final List<Cursor> viewResult = domainResultCursors.stream().map(cursorConverter::convert)
+                .collect(Collectors.toList());
+
+        return status(OK).body(viewResult);
+    }
+
+    @RequestMapping(path = "/event-types/{eventTypeName}/cursors-lag", method = RequestMethod.POST)
+    public ResponseEntity<?> cursorsLag(@PathVariable("eventTypeName") final String eventTypeName,
+                                         @Valid @RequestBody final List<Cursor> cursors,
+                                         final Errors errors,
+                                         final NativeWebRequest request) {
+
+        if (errors.hasErrors()) {
+            return Responses.create(new ValidationProblem(errors), request);
+        }
```

Comment:
Completely removed this code since it's already handled automatically by another exception handler. It even provides a very nicely formatted error message explaining the reason and the path for each error

`{"violations":[{"field":"valid_list_wrapper.list[0].offset","message":"may not be null"}],"status":422,"title":"Constraint Violation","type":"https://github.com/zalando/problem/wiki/constraint-violation"}`